### ISSUE TITLE
Some generated siva files have no packfiles

### ISSUE THREAD

**Issue by erizocosmico** [src-d/borges#221]
I'm processing all fetched repositories in the apps postgres (`/apps/borges/root-repositories` in HDFS) and I've come across a few errors while doing so.

Some repositories say that `object is no found` when trying to get the HEAD commit and others that there is no HEAD at all.

I did go to HDFS and download one of these corrupted siva files and did a `siva unpack`. It had `HEAD`,  `config`, `refs` and no `objects` (and thus no packfile) so it checks out.

Here is a log with all the IDs of these repositories so far (there are more) https://gist.github.com/erizocosmico/92403aaf7569566431a7ddb26b2b4d79 along with the error they gave

**Comment by erizocosmico** [comment]
UPDATE: i tried one of them and this was the output

```
EROR[02-05|18:03:08] error cloning repository                 module=borges worker=3 job=016166e8-4fb3-12bf-fb4e-861153fc23ce endpoint=http://github.com/urbn/URBNAlert error="empty git-upload-pack given" caller=archiver.go:163```

**Comment by ajnavarro** [comment]
the error `repository has no HEAD` is because the repository is empty.

**Comment by erizocosmico** [comment]
I cannot reproduce this with all the repositories I've tried from the list. Shall we close this if we can't reproduce it or leave it open to keep it in mind?

**Comment by ajnavarro** [comment]
Can be related with https://github.com/src-d/engine/issues/336

**Comment by ajnavarro** [comment]
The problem appears to be here:

https://github.com/src-d/borges/blob/master/archiver.go#L357

The configuration changes are made using the rooted repo directly, instead to use the transactioner, but I cannot localize the not handled error that is causing the commit of the rooted repo without doing the `Push` of the new packfile with the new objects.

**Comment by jfontan** [comment]
This is probably caused because packfile and index is written when `PackWriter` is closed:

https://github.com/src-d/go-git/blob/master/storage/filesystem/internal/dotgit/writers.go#L93

There is a bug that discards close errors in go-git. Already opened an issue there:

https://github.com/src-d/go-git/issues/791

**Comment by jfontan** [comment]
Fixed with src-d/go-git#791
